packages/vm/src/modules/file_system_cache.rs

```diff
@@ -80,27 +92,17 @@ impl FileSystemCache {
         let mmap = unsafe { Mmap::map(&file) }
             .map_err(|e| VmError::cache_err(format!("Mmap error: {}", e)))?;
 
-        let serialized_cache = Artifact::deserialize(&mmap[..])?;
-        let module = unsafe {
-            wasmer_runtime_core::load_cache_with(
-                serialized_cache,
-                compiler_for_backend(backend)
-                    .ok_or_else(|| VmError::cache_err(format!("Unsupported backend: {}", backend)))?
-                    .as_ref(),
-            )
-        }?;
+        let store = make_store_headless(memory_limit);
```

Comment:
How does this compare with the other Cache above?

It seems that you can set the memory limit on loading a cached Module from disk, not just compile-time, which is a nice flexibility (eg. could change when restarting wasmd without requiring a recompile of all modules).

Reply:
Ah, I see `src/cache.rs` is using this as a building block, here is where the memory limit is being enforced